addons/point_of_sale/point_of_sale.py

```diff
@@ -39,6 +39,45 @@ def _get_currency(self, cr, uid, ids, fieldnames, args, context=None):
             result[pos_config.id] = currency_id
         return result
 
+    def _get_current_session(self, cr, uid, ids, fieldnames, args, context=None):
+        result = dict()
+
+        for record in self.browse(cr, uid, ids, context=context):
+            session_id = record.session_ids.filtered(lambda r: r.user_id.id == uid and not r.state == 'closed')
+            result[record.id] = {
+                'current_session_id': session_id,
+                'current_session_state': session_id.state,
+            }
+        return result
+
+    def _get_last_session(self, cr, uid, ids, fieldnames, args, context=None):
+        result = dict()
+
+        for record in self.browse(cr, uid, ids, context=context):
+            session_ids = self.pool['pos.session'].search_read(
+                cr, uid,
+                [('config_id', '=', record.id), ('state', '=', 'closed')],
+                ['cash_register_balance_end_real', 'stop_at'],
+                order="stop_at desc", limit=1, context=context)
+            if session_ids:
+                result[record.id] = {
+                    'last_session_closing_cash': session_ids[0]['cash_register_balance_end_real'],
+                    'last_session_closing_date': session_ids[0]['stop_at'],
+                }
+            else:
+                result[record.id] = {
+                    'last_session_closing_cash': 0,
+                    'last_session_closing_date': None,
+                }
+        return result
+
+    def _get_current_user(self, cr, uid, ids, fieldnames, args, context=None):
+        result = dict()
+
+        for record in self.browse(cr, uid, ids, context=context):
+            result[record.id] = record.session_ids.filtered(lambda r: r.state == 'opened').user_id.name
```

Comment:
Same as above, what happens when there is no session_ids which matches the filter ?